src/software/ai/navigator/navigator.h

```diff
@@ -112,29 +121,33 @@ class Navigator : public IntentVisitor
     // This variable is set by each `visit` function
     std::unique_ptr<Primitive> current_primitive;
 
-    // The current Robot the navigator has navigated for from an Intent.
-    // This variable is set by each `visit` function
-    std::optional<Robot> current_robot;
+    /**
+     * These are obstacles that represent robots that aren't
+     * assigned move intents
+     * When move intents are processed to path plan,
+     * we can avoid these non-"moving" robots
+     */
+    std::set<RobotId> non_path_planning_robots;
 
-    // The current destination the navigator has navigated to from an Intent.
-    // This variable is set by each `visit` function
-    Point current_destination;
+    // This is used by the visualizer to see the planned paths
+    std::vector<std::vector<Point>> planned_paths;
 
-    std::vector<Obstacle> velocity_obstacles;
+    // path objectives are used to plan paths
+    std::map<RobotId, PathObjective> robot_id_to_path_objectives;
```

Comment:
nit; but we're mapping `robot_id` to a single path objective, so this should probably be `robot_id_to_path_objective`

Reply:
done